### PR TITLE
#170045960 build the signup endpoint with database

### PR DESCRIPTION
**Pull request by Nigorjeanluc**
### What does this PR do?
This pull request set up the route to create new user's account and save data in a database

### Description of Task to be completed?
The tasks done here are
- add validation to this endpoint for v2
- add models and integrate DB to save user's data
- add a method for the API to handle this endpoint

### How should this be manually tested?
Clone or download the develop branch to your local machine and you can see the changes.

### What are the relevant pivotal tracker stories?
[#170045960](https://www.pivotaltracker.com/n/projects/2410710/stories/170045960)

### CODE REVIEW

File: server/v2/controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        try {
+            const existingEmail = await pool.query(queries.findOneUser, [req.body.email]);
+            if (existingEmail.rowCount === 1) {
+                return res.status(409).json({
+                    status: 409,
+                    error: 'Email already exists!'
+                });
+            }
+
+            const existingUsername = await pool.query(queries.findOneUsername, [req.body.username]);
```
Comment (mwibutsa) [inline server/v2/controllers/userController.js]: Move database related operations to the relevant Model; for instance, the methods to create, update, delete and get user info should all be located in user Model.

File: server/v2/tests/auth.spec.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        Chai.request(app)
+            .post("/api/v2/auth/signup")
+            .send(usersTester[0])
+            .end((err, res) => {
+                res.should.have.status(201);
+                res.body.should.have.property("message", "User created successfully");
+                res.body.should.be.a("object");
+                done();
+            });
+    });
```
Comment (mwibutsa) [inline server/v2/tests/auth.spec.js]: As we have all agreed you should be using async-await in your tests. it is the Modern way of writing tests and is preferred by most developers than using callbacks.

File: server/v2/tests/auth.spec.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            .send(usersTester[3])
+            .end((err, res) => {
+                res.should.have.status(409);
+                res.body.should.have.property(
+                    "error",
+                    "Username already exists!"
+                );
+                done();
+            });
+    });
```
Comment (mwibutsa) [inline server/v2/tests/auth.spec.js]: This also needs to use async await

**Review by mwibutsa — CHANGES_REQUESTED** [review]

**Comment by mwibutsa** [comment]
if there is a way you can keep your tests coverage to 100% it would be better but make sure you don't go bellow 95%

File: server/v2/tests/auth.spec.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    it("should create a new user account",
+        async() => {
+            const res = await Chai.request(app)
+                .post("/api/v2/auth/signup")
+                .send(usersTester[0]);
+            expect(res.body.status).to.equal(201);
+            expect(res.body).to.be.an('object');
+            expect(res.body.message).to.be.a('string');
+        });
```
Comment (mwibutsa) [inline server/v2/tests/auth.spec.js]: I'm glad that you are using async-await in your tests. Usually, when you are using async-await you need to nest the block inside a try-catch block to capture the errors.  that's the only thing which is missing in your test.

**Review by mwibutsa — CHANGES_REQUESTED** [review]
Even though you are writing tests you need to handle the case in which there might be an error. consider using nesting any async await code inside a ```try catch```  block.

**Comment by mwibutsa** [comment]
@Nigorjeanluc, Don't forget to squash your commits into one. it looks more professional when they are squashed.

File: server/v2/tests/auth.spec.js (hunk 4 above)
Comment (Nigorjeanluc) [reply server/v2/tests/auth.spec.js]: OK, I'm going to work on it according to the resource you gave us yesterday. Thanks again

**Review by mwibutsa — APPROVED** [review]
This work looks amazing @Nigorjeanluc

**Pull request merged**